app/ui-react/packages/ui/src/utils/testIdGenerator.ts

```diff
@@ -1,30 +1,31 @@
 /**
- * Generates an identifier suitable to use as a `data-testid`. The prefix is separated by a dot character.
- * @param prefix text that will be used as a prefix (typically the component's class name)
- * @param uniqueId unique text within a prefix
- * @returns a generated test identifier
+ * Generates an identifier suitable to use as a `data-testid`. Value arguments are separated by two
+ * dash characters.
+ * @param values the values used to generate a test identifier
+ * @returns a test identifier
  */
-export function toTestId(prefix: string, uniqueId: string): string {
-  if (prefix && uniqueId) {
-    return generateId(prefix) + '--' + generateId(uniqueId);
-  }
+export function toTestId(...values: string[]): string {
+  let testId = '';
 
-  if (prefix) {
-    return generateId(prefix);
-  }
+  for (let i = 0; i < values.length; i++) {
```

Comment:
Couldn't you do `values.map(generateId).join('-')` instead of this `for` loop etc?

Reply:
I'll make that change. This changes the segment separator from two dashes to one dash but that is no big deal as we talked about @gashcrumb.